flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/StreamSource.java

```diff
@@ -299,7 +319,7 @@ public void trigger(long timestamp) {
 					synchronized (lockingObject) {
 						if (currentTime > nextWatermarkTime) {
 							output.emitWatermark(new Watermark(watermarkTime));
-							nextWatermarkTime += watermarkInterval;
+							nextWatermarkTime = watermarkTime + watermarkInterval;
```

Comment:
I do not think it does because the nextWatermarkTime is the previous + watermarkInterval. Given that we just emitted a watermark with {{watermarkTime}} timestamp, the next should be watermarkTime + interval. In addition, the previous version could create issues especially with testing, as the nextWatermark variable is not initialized so it takes the value 0L and this led to the same first watermark being emitted twice, once in the trigger and once in the collect() above, which would set the correct nextWatermarkTime.